android/src/main/java/com/onesignal/flutter/FlutterRegistrarResponder.java

```diff
@@ -54,7 +58,12 @@ public void run() {
    }
 
    private void runOnMainThread(final Runnable runnable) {
-      ((Activity) flutterRegistrar.activeContext()).runOnUiThread(runnable);
+      if (Looper.getMainLooper().getThread() == Thread.currentThread())
+         runnable.run();
+      else {
+         Handler handler = new Handler(Looper.getMainLooper());
+         handler.post(runnable);
+      }
```

Comment:
Looks good, as discussed in #190 this `runOnMainThread` method now matches the exact native Android SDK implementation